### PR TITLE
WIP Multimodal support

### PR DESCRIPTION
**Pull request by winglian**

### CODE REVIEW

File: src/axolotl/utils/models.py [hunk 1]
```diff
@@ -298,10 +305,28 @@ def load_tokenizer(cfg):
         )
     return tokenizer
 
+def load_processor(cfg: DictDefault, tokenizer: PreTrainedTokenizerBase):
+    #model_config = load_model_config(cfg)
+    processor_kwargs = {} #do we actually need this?
+
+    processor_cls = AutoProcessor
+    if cfg.processor_type:
```
Comment (winglian) [inline src/axolotl/utils/models.py:312]: need to add `processor_type` to https://github.com/axolotl-ai-cloud/axolotl/blob/main/src/axolotl/utils/config/models/input/v0_4_1/__init__.py

File: src/axolotl/train.py [hunk 2]
```diff
@@ -69,6 +69,9 @@ def train(
         main_process_only=True,
     )
     tokenizer = load_tokenizer(cfg)
+    processor = None
+    if cfg.is_multimodal:
```
Comment (winglian) [inline src/axolotl/train.py:75]: also need to include `is_multimodal` arg to the config class

Comment (winglian) [inline src/axolotl/train.py:75]:
```suggestion
    if cfg.is_multimodal or cfg.processor_type:
```

File: src/axolotl/utils/models.py [hunk 3]
```diff
@@ -461,6 +491,14 @@ def load_model(
     max_memory = cfg.max_memory
     device_map = cfg.device_map
 
+    if cfg.is_multimodal:
+        if model_config.model_type == "llava":
+            Auto = LlavaForConditionalGeneration
+        elif model_config.model_type == "mllama":
+            Auto = MllamaForConditionalGeneration
```
Comment (winglian) [inline src/axolotl/utils/models.py]: we can use `AutoModelForVision2Seq` instead here.

**Comment by bursteratom** [comment]
this is too outdated. Gonna start a new one.

**Pull request closed without merging**